File: src/main/java/arc/arc/invest/Production.java
package arc.arc.invest;

import arc.arc.invest.goods.ProductionEntry;
import lombok.RequiredArgsConstructor;
import org.bukkit.configuration.ConfigurationSection;
import org.bukkit.inventory.ItemStack;

import java.util.*;

@RequiredArgsConstructor
public class Production {

    final List<ProductionEntry> productionEntries;

    public List<ItemStack> produce(Set<String> productionIds, int level) {
        List<ItemStack> stacks = new ArrayList<>();
        for(ProductionEntry entry : productionEntries){
            if(!productionIds.contains(entry.getName())) continue;
            stacks.addAll(entry.production(level));
        }
        return stacks;
    }

    public Map<String, List<ItemStack>> reqs(int level){
        Map<String, List<ItemStack>> reqs = new HashMap<>();
        for(var entry : productionEntries){
            reqs.put(entry.getName(), entry.reqs(level));
        }
        return reqs;
    }

    public static Production deserialize(ConfigurationSection section) {
        if(section == null) return null;
        List<ProductionEntry> productionEntryList = new ArrayList<>();
        for (String key : section.getKeys(false)) {
            ProductionEntry en = ProductionEntry.deserialize(section.getConfigurationSection(key), key);
            productionEntryList.add(en);
        }
        return new Production(productionEntryList);
    }

}

File: src/main/kotlin/ru/arc/invest/Production.kt
package ru.arc.invest

import org.bukkit.configuration.ConfigurationSection
import org.bukkit.inventory.ItemStack
import ru.arc.invest.goods.ProductionEntry

class Production(private val productionEntries: List<ProductionEntry>) {

    fun produce(productionIds: Set<String>, level: Int): List<ItemStack> {
        val stacks = ArrayList<ItemStack>()
        for (entry in productionEntries) {
            if (!productionIds.contains(entry.name)) continue
            stacks.addAll(entry.production(level))
        }
        return stacks
    }

    fun reqs(level: Int): Map<String, List<ItemStack>> {
        val reqs = HashMap<String, List<ItemStack>>()
        for (entry in productionEntries) {
            reqs[entry.name] = entry.reqs(level)
        }
        return reqs
    }

    companion object {
        @JvmStatic
        fun deserialize(section: ConfigurationSection?): Production? {
            section ?: return null
            val productionEntryList = ArrayList<ProductionEntry>()
            for (key in section.getKeys(false)) {
                val en = ProductionEntry.deserialize(section.getConfigurationSection(key)!!, key)
                productionEntryList.add(en)
            }
            return Production(productionEntryList)
        }
    }
}
